build_runner/lib/src/entrypoint/test.dart

```diff
@@ -154,6 +156,22 @@ void _ensureBuildTestDependency(PackageGraph packageGraph) {
   }
 }
 
+void _ensureProcessExit(Process process) {
+  var signalsSub = _exitProcessSignals.listen((signal) async {
+    stdout.writeln('waiting for subprocess to exit...');
+    await process.exitCode;
```

Comment:
@natebosch brought up that we actually don't need this await here either :)

Reply:
Yeah I think this can be:

```dart
var signalsSub = _exitProcessSignals.listen((_) {
    stdout.writeln('waiting for subprocess to exit...');
});
```

Reply:
Fixed in 9c1707f